### ISSUE TITLE
Implement Sparsemax: multilabel classification tuned for attention based neural network

### ISSUE THREAD

**Issue by mratsim** [mratsim/Arraymancer#285]
Instead of always assuming label independence by using a binary relevance (aka reduction of individual binary cross-entropies), it would be nice to have loss functions made for multilabeling problems.

Sparsemax seems promising:

- Paper: [[1602.02068]](https://arxiv.org/pdf/1602.02068.pdf)
- [Tensorflow implementation](https://github.com/tensorflow/tensorflow/blob/3ab16ebce6a0a9ce20120c3c2dd1f1a8cf5b2ad8/tensorflow/contrib/sparsemax/python/ops/sparsemax.py)
- [PyTorch implementation](https://github.com/msobroza/SparsemaxPytorch)
- [Lua Torch implementation](https://github.com/gokceneraslan/SparseMax.torch)
